Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1104,6 +1105,44 @@ def print_log(self, verbose=False):
             demisto.info(text)
             self.messages = []
 
+    def build_curl(self, text):
+        http_methods = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH']
+        data = text.split("send: b'")[1]
+        if data.startswith('{'):
+            # it is the request url query params/post body - will always come after we already have the url and headers
+            self.curl[-1] += "-d '{}".format(data)
+        elif any(http_method in data for http_method in http_methods):
+            method = ''
+            url = ''
+            headers = []
+            headers_to_skip = ['Content-Length', 'User-Agent', 'Accept-Encoding', 'Connection']
+            request_parts = repr(data).split('\\\\r\\\\n')  # splitting lines on repr since data is a bytes-string
+            for line, part in enumerate(request_parts):
+                if line == 0:
+                    method, url, _ = part[1:].split()  # ignoring " at first char
+                elif line != len(request_parts) - 1:  # ignoring the last line which is empty
+                    if part.startswith('Host:'):
+                        _, host = part.split('Host: ')
+                        url = 'https://{}{}'.format(host, url)
+                    else:
+                        if any(header_to_skip in part for header_to_skip in headers_to_skip):
+                            continue
+                        headers.append(part)
+            curl_headers = ''
+            for header in headers:
+                if header:
+                    curl_headers += '-H "{}" '.format(header)
+            curl = 'curl -X {} {} {}'.format(method, url, curl_headers)
+            if demisto.params().get('proxy'):
+                proxy_address = os.environ.get('https_proxy')
+                if proxy_address:
+                    curl += '--proxy {} '.format(proxy_address)
+            else:
+                curl += '--noproxy '
+            if demisto.params().get('insecure'):
+                curl += '-k '
+            self.curl.append(curl)
```

Comment:
Why not just print it out to the log? Why do we need to store this? It may get big and also it is better to have it next to when the call happened.

Reply:
@anara123 suggested the same and i think it will bring noise as it will break the flow of the log where i can see the what was sent and what returned 
also, if there is more than one call, i'd prefer to see all the curls one after the other so it would be easier to read them

Reply:
Ok. We can try this way.